dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -45,393 +79,500 @@ public class DidDhtConfiguration internal constructor(
+                val valueItems = values.split(",")
 
-                // add verification relationships
-                doc.authenticationVerificationMethods(lists["auth"])
-                doc.assertionMethodVerificationMethods(lists["asm"])
-                doc.keyAgreementVerificationMethods(lists["agm"])
-                doc.capabilityInvocationVerificationMethods(lists["inv"])
-                doc.capabilityDelegationVerificationMethods(lists["del"])
+                valueItems.forEach {
+                  lists[key]?.add(VerificationMethod.builder().id(URI(keyLookup[it]!!)).build())
+                }
               }
+
+              // add verification relationships
+              doc.authenticationVerificationMethods(lists["auth"])
+              doc.assertionMethodVerificationMethods(lists["asm"])
+              doc.keyAgreementVerificationMethods(lists["agm"])
+              doc.capabilityInvocationVerificationMethods(lists["inv"])
+              doc.capabilityDelegationVerificationMethods(lists["del"])
             }
           }
         }
       }
+    }
 
-      // add verification methods and services
-      doc.verificationMethods(verificationMethods)
-      doc.services(services)
+    // add verification methods and services
+    doc.verificationMethods(verificationMethods)
+    doc.services(services)
 
-      return doc.build() to types
-    }
+    return doc.build() to types
+  }
 
-    /**
-     * Parses a string of key-value pairs separated by semicolons into a map.
-     * @param data The string to parse.
-     */
-    private fun parseTxtData(data: String): Map<String, String> {
-      return data.split(";").associate {
-        val (key, value) = it.split("=")
-        key to value
-      }
+  /**
+   * Parses a string of key-value pairs separated by semicolons into a map.
+   * @param data The string to parse.
+   */
+  private fun parseTxtData(data: String): Map<String, String> {
+    return data.split(";").associate {
+      val (key, value) = it.split("=")
+      key to value
     }
   }
+}
+
+/**
+ * Provides a specific implementation for creating and resolving "did:dht" method Decentralized Identifiers (DIDs).
+ *
+ * A "did:dht" DID is a special type of DID that is based on an identity key, but can be extended to contain other
+ * keys, services, and other DID Document properties. It relies upon Distributed Hash Table (DHT) provided by the
+ * BitTorrent network, and an intermediary layer called Pkarr (Public Key Addressable Records) to store and retrieve
+ * the DID Document.
+ *
+ * @property uri The URI of the "did:dht" which conforms to the DID standard.
+ * @property keyManager A [KeyManager] instance utilized to manage the cryptographic keys associated with the DID.
+ * @property didDocument The [DIDDocument] associated with the DID, created by the class.
+ */
+public class DidDht(
+  uri: String, keyManager: KeyManager,
+  public val didDocument: DIDDocument? = null,
```

Comment:
Curious why this can be nullable? Is it possible to make it non-nullable?

Reply:
because you can construct an instance of the class before having a DID Document